File: 3332-minimum-operations-to-exceed-threshold-value-ii/3332-minimum-operations-to-exceed-threshold-value-ii.py
class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        heapq.heapify(nums)
        op = 0
        while (True):
            p = nums[0]
            if ( p >= k or len(nums) == 1):
                break
            sm1 = heapq.heappop(nums)
            sm2 = heapq.heappop(nums)
            heapq.heappush(nums , (min(sm1 , sm2) * 2)+max(sm1,sm2))
            op += 1
        return op
        

File: 3332-minimum-operations-to-exceed-threshold-value-ii/3332-minimum-operations-to-exceed-threshold-value-ii.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int k) {
        priority_queue<long long , vector<long long> , greater<long long>> pq;
        for ( auto i : nums){
            pq.push(i);
        }
        int op = 0;
        long long p , sm1 , sm2; 
        while(true){
            p  = pq.top();
            if (p>=k || pq.size() == 1){
                break;
            }
            sm1 = pq.top();
            pq.pop();
            sm2 = pq.top();
            pq.pop();
            pq.push(min(sm1 , sm2) * 2 + max(sm1 , sm2));
            op ++;
        }
        return op;

    }
};